native/src/gpxRouteApproximation.cpp

```diff
@@ -0,0 +1,524 @@
+#include "gpxRouteApproximation.h"
+
+#include "routeResultPreparation.h"
+#include "routePlannerFrontEnd.h"
+#include "binaryRoutePlanner.h"
+#include "routeCalculationProgress.h"
+
+// setRouter() must be called by router before approximation
+// both constructors are saved "as is" for iOS compatibility
+void GpxRouteApproximation::setRouter(RoutePlannerFrontEnd* router) {
```

Comment:
Move method below constructors

Reply:
In this particular case, setRouter() is more important than constructors. Ok, moved below.